url/src/comp.php

```diff
@@ -361,35 +361,35 @@ function delete_url($itemID, $oldURL, $shortCode, $TalisGUID, $token, ReportRow
 				$online_resource_found = true;
 			}
 
+			$body = get_patch_template($resource_id);
+
 			// if we do have web addresses that we can work with...
 			if ($web_address_array){
 				$web_address_found = true;
 				$new_web_address_array = check_web_addresses($oldURL, "", $web_address_array, "delete");
 				$report->newWebAddressArray = $new_web_address_array;
 
-				$body = get_patch_template($resource_id);
-
 				// update the web addresses if they have changed
 				if($web_address_array !== $new_web_address_array){
 					$body = patch_web_addresses($body, $new_web_address_array);
 					$actionMessagePart[] = 'URL Deleted Successfully';
 				}
+			}
 
-				// remove online resource if it matches the URL to remove.
-				// If this is the secondary - we never want to have an online resource set there
-				if ($online_resource === $oldURL || ($primary_resource !== $resource_id && $online_resource_found === true)) {
-					$body = patch_online_resource($body, null);
-					$actionMessagePart[] = 'Had to remove online resource';
-				}
-
-				// if no changes have been made to the template then there is nothing to do.
-				if ($body === get_patch_template($resource_id)) {
-					echo_message_to_screen(DEBUG, "Nothing to update with this resource</br>");
-					$report->actionMessage = "Nothing to do";
-					$report->updated = true;
-					continue;
-				}
+			// remove online resource if it matches the URL to remove.
+			// If this is the secondary - we never want to have an online resource set there
+			if ($online_resource === $oldURL || ($primary_resource !== $resource_id && $online_resource_found === true)) {
+				$body = patch_online_resource($body, null);
+				$actionMessagePart[] = 'Had to remove online resource';
+			}
 
+			// if no changes have been made to the template then there is nothing to do.
+			if ($body === get_patch_template($resource_id)) {
```

Comment:
The update will now happen regardless of whether either the online resource or web address or both have been updated